internal/provider/providers/hetzner/create.go

```diff
@@ -0,0 +1,84 @@
+package hetzner
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/netip"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/provider/constants"
+	"github.com/qdm12/ddns-updater/internal/provider/errors"
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+)
+
+func (p *Provider) createRecord(ctx context.Context, client *http.Client, ip netip.Addr) (err error) {
+	recordType := constants.A
+	if ip.Is6() {
+		recordType = constants.AAAA
+	}
+
+	u := url.URL{
+		Scheme: "https",
+		Host:   "dns.hetzner.com",
+		Path:   fmt.Sprintf("/api/v1/records"),
+	}
+
+	requestData := struct {
+		Type           string `json:"type"`
+		Name           string `json:"name"`
+		Value          string `json:"value"`
+		ZoneIdentifier string `json:"zone_id"`
+		TTL            uint   `json:"ttl"`
+	}{
+		Type:           recordType,
+		Name:           p.host,
+		Value:          ip.String(),
+		ZoneIdentifier: p.zoneIdentifier,
+		TTL:            p.ttl,
+	}
+
+	buffer := bytes.NewBuffer(nil)
+	encoder := json.NewEncoder(buffer)
+	err = encoder.Encode(requestData)
+	if err != nil {
+		return fmt.Errorf("JSON encoding request data: %w", err)
+	}
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), buffer)
+	if err != nil {
+		return fmt.Errorf("creating http request: %w", err)
+	}
+
+	p.setHeaders(request)
+
+	response, err := client.Do(request)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode > http.StatusUnsupportedMediaType {
+		return fmt.Errorf("%w: %d: %s",
+			errors.ErrHTTPStatusNotValid, response.StatusCode, utils.BodyToSingleLine(response.Body))
+	}
+
+	decoder := json.NewDecoder(response.Body)
+	var parsedJSON struct {
+		Record struct {
+			ID string `json:"id"`
+		} `json:"record"`
```

Comment:
Can we also get the `"value"` field so that we can return the `newIP` from `createRecord`, as returned from the server? So that we can compare it with the ip we sent. A bit of a extreme cautionary measure, but better safe than sorry 😉